nixos/modules/programs/rush.nix

```
@@ -0,0 +1,127 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.programs.rush;
+  opt = options.programs.rush;
+
+  indent =
+    lines:
+    lib.pipe lines [
+      (lib.splitString "\n")
+      (builtins.filter (line: line != ""))
+      (map (line: "  " + line))
+      (builtins.concatStringsSep "\n")
+    ];
+in
+{
+  meta.maintainers = with lib.maintainers; [
+    c4f3z1n
+  ];
+
+  options.programs.rush = with lib.types; {
+    enable = lib.mkEnableOption "Whether to enable Restricted User Shell." // {
+      default = opt.global.isDefined || opt.rules.isDefined;
```

```suggestion
    enable = lib.mkEnableOption "Restricted User Shell";
```
1. `mkEnableOption` already prepends "whether to enable".
2. Do not enable the module based on the config existence, modules should be opt-in.